examples/advance_music_cog.py

```diff
@@ -0,0 +1,457 @@
+                name = 'Played',
+                value = parse_duration(duration_played)
+                )
+            embed.add_field(
+                name = 'Duration',
+                value = parse_duration(player.duration)
+                )
+            embed.add_field(
+                name = 'Loop',
+                value = loop_status
+                )
+            embed.add_field(
+                name = 'Requested by',
+                value = player.requester.mention
+                )
+            embed.add_field(
+                name = 'Uploader',
+                value = uploader
+                )
+            embed.add_field(
+                name = 'URL',
+                value = f'[Click]({url})'
+                )
+            embed.add_field(
+                name = 'Views',
+                value = parse_count(int(views))
+                )
+            embed.add_field(
+                name ='Rating',
+                value = rating
+                )
+            embed.set_thumbnail(
+                url = thumbnail
+                )
+            embed.set_image(
+                url = r"https://i.imgur.com/ufxvZ0j.gif"
+                )
+            embed.set_author(
+                name = ctx.author.name,
+                icon_url = ctx.author.avatar_url
+                )
+
+            #sending the embed
+            await ctx.send(embed=embed)
+
+    #join command        
+    @commands.command()
+    async def join(self, ctx):
+        if await self.MusicManager.join(ctx):
+            await ctx.send("Joined Voice Channel")
+
+    #play command
+    @commands.command()
+    async def play(self, ctx, *, query: str):
+        #checking if the bot has joined a voice channel
+        if not ctx.voice_client or not ctx.voice_client.is_connected():
+            await self.MusicManager.join(ctx)
+
+        #searching while showing typing status
+        async with ctx.typing(): 
+            players = await self.MusicManager.create_player(query, ctx.author)
+        #^change to async ctx.defer() if using slash commands
+
+        #if a player object is created ie. song found
+        if players:
+            if await self.MusicManager.queue_add(
+                players=players, 
+                ctx=ctx
+            ) and not await self.MusicManager.play(ctx):
+                #sending a message
+                await ctx.send(f"Added {players[0].title} to song queue.")
+
+        else:
+            await ctx.send("Query not found.")
+
+    #pause command
+    @commands.command()
+    async def pause(self, ctx):
+        if await self.MusicManager.pause(ctx):
+            await ctx.send("Paused")
+
+    #resume command
+    @commands.command()
+    async def resume(self, ctx):
+        if await self.MusicManager.resume(ctx):
+            await ctx.send("Resumed")
+
+    #volume command
+    @commands.command()
+    async def volume(self, ctx, volume: int):
+        await self.MusicManager.volume(ctx, volume)
```

Comment:
What if the bot is not connected?
```py
if volume := await self.MusicManager.volume(ctx, volume) is not None:
    ...
 ```

Reply:
For the bot voice channel connection, I have made a before_invoke check. I'll add all these commands to the check

Reply:
You should not do that, and instead, use the default ones DSU provides as they might make issues.